components/script/dom/console.rs

```diff
@@ -55,14 +64,138 @@ where
+                        return value_to_source(cx, value);
+                    }
+
+                    let mut ids = IdVector::new(cx);
+                    if !GetPropertyKeys(
+                        cx,
+                        obj.handle().into(),
+                        jsapi::JSITER_OWNONLY | jsapi::JSITER_SYMBOLS,
+                        ids.handle_mut(),
+                    ) {
+                        return DOMString::from("/* invalid */");
+                    }
+                    let mut explicit_keys = object_class == ESClass::Object;
+                    let mut props = Vec::with_capacity(ids.len());
+                    for id in &*ids {
+                        rooted!(in(cx) let id = *id);
+                        rooted!(in(cx) let mut desc = PropertyDescriptor::default());
+
+                        let mut is_none = false;
+                        if !JS_GetOwnPropertyDescriptorById(
+                            cx,
+                            obj.handle().into(),
+                            id.handle().into(),
+                            desc.handle_mut().into(),
+                            &mut is_none,
+                        ) {
+                            return DOMString::from("/* invalid */");
+                        }
+
+                        rooted!(in(cx) let mut property = UndefinedValue());
+                        if !JS_GetPropertyById(cx, obj.handle(), id.handle(), property.handle_mut())
+                        {
+                            return DOMString::from("/* invalid */");
+                        }
+
+                        if !explicit_keys {
+                            if id.is_int() {
+                                if let Ok(id_int) = usize::try_from(id.to_int()) {
+                                    explicit_keys = props.len() != id_int;
+                                } else {
+                                    explicit_keys = false;
+                                }
+                            } else {
+                                explicit_keys = false;
+                            }
+                        }
+                        let value_string = stringify_inner(cx, property.handle(), parents.clone());
+                        if explicit_keys {
+                            let key = if id.is_string() || id.is_symbol() || id.is_int() {
+                                rooted!(in(cx) let mut key_value = UndefinedValue());
+                                let raw_id: jsapi::HandleId = id.handle().into();
+                                if !JS_IdToValue(cx, *raw_id.ptr, key_value.handle_mut()) {
+                                    return DOMString::from("/* invalid */");
+                                }
+                                value_to_source(cx, key_value.handle())
+                            } else {
+                                panic!("Key with impossible type")
```

Comment:
Instead of panicking, please just return an error string like you do with "/* invalid */" I don't think we want to stop the entire script thread for this.